### PR TITLE
[serverless] Support trace context propagation for ALB target groups with MultiValueHeaders

### PR DESCRIPTION
**Pull request by purple4reina**
Originally https://github.com/DataDog/datadog-agent/pull/30764

### What does this PR do?

Adds support for extracting trace context from MultiValueHeaders in ALBTargetGroupRequest

### Motivation

Currently, Datadog APM context propagation doesn't work when MultiValueHeaders is enabled in ALB target groups. This PR resolves this issue by allowing trace context propagation even when using MultiValueHeaders configuration.

### Describe how to test/QA your changes

- Enable MultiValueHeaders in ALB target group
- Create a Lambda endpoint
- Send a request with trace context headers to the ALB endpoint
- Verify in Datadog's trace view that the trace context is properly propagated from ALB to Lambda

### Possible Drawbacks / Trade-offs

- Only the first value from MultiValueHeaders is used, subsequent values are ignored

### Additional Notes

- This PR includes test coverage:
  - Unit tests for MultiValueHeadersCarrier
  - Tests for MultiValueHeaders extraction in ALB target groups
  - Tests for MultiValueHeaders support in tag extraction functionality
- Maintains backward compatibility by prioritizing the existing Headers field

https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#multi-value-headers-request

### CODE REVIEW

File: pkg/serverless/trace/propagation/carriers.go [hunk 1]
```diff
@@ -253,6 +253,16 @@ func headersCarrier(hdrs map[string]string) (tracer.TextMapReader, error) {
...
 }
 
+// headersOrMultiheadersCarrier returns the tracer.TextMapReader used to extract
+// trace context from a Headers field of form map[string]string or MultiValueHeaders
+// field of form map[string][]string.
+func headersOrMultiheadersCarrier(hdrs map[string]string, multiHdrs map[string][]string) (tracer.TextMapReader, error) {
+	if len(hdrs) > 0 {
+		return headersCarrier(hdrs)
+	}
+	return tracer.HTTPHeadersCarrier(multiHdrs), nil
```
Comment (nhulston) [inline pkg/serverless/trace/propagation/carriers.go:263]: Just a question -- why do we use the `TextMapCarrier` and `HTTPHeadersCarrier` from the Go tracer? Seems odd that the Go tracer is a dependency of the agent

Comment (purple4reina) [reply pkg/serverless/trace/propagation/carriers.go:263]: We rely on the tracer for trace context extraction. We create one of these "carrier" objects, then pass it to the `Propagator.Extract` method defined by the tracer. By relying on the tracer to do this work, it means we don't need to implement the complex logic to do it ourselves.

**Review by nhulston — APPROVED** [review]
Looks good! Created a simple test app to test, and the trace propagation is working as expected. We see my custom trace ID being propagated: https://ddserverless.datadoghq.com/api/v1/trace/1122334455?graphType=flamegraph&shouldShowLegend=true&sort=time&spanID=753545956937652535&timeHint=1733511285145.049

**Pull request merged**